src/Tasks/Microsoft.NET.Build.Tasks/CompileReadyToRunImages.cs

```diff
@@ -0,0 +1,376 @@
+            try
+            {
+                using (FileStream fs = new FileStream(file.ItemSpec, FileMode.Open, FileAccess.Read))
+                {
+                    PEReader pereader = new PEReader(fs);
+
+                    if (pereader.PEHeaders == null || pereader.PEHeaders.CorHeader == null)
+                        return false;
+
+                    if ((pereader.PEHeaders.CorHeader.Flags & CorFlags.ILOnly) != CorFlags.ILOnly)
+                        return false;
+
+                    // Skip assemblies that reference winmds
+                    MetadataReader mdReader = pereader.GetMetadataReader();
+                    foreach (var assemblyRefHandle in mdReader.AssemblyReferences)
+                    {
+                        AssemblyReference assemblyRef = mdReader.GetAssemblyReference(assemblyRefHandle);
+                        if ((assemblyRef.Flags & AssemblyFlags.WindowsRuntime) == AssemblyFlags.WindowsRuntime)
+                            return false;
+                    }
+
+                    // Skip reference assemblies
+                    foreach (var attributeHandle in mdReader.GetAssemblyDefinition().GetCustomAttributes())
+                    {
+                        EntityHandle attributeCtor = mdReader.GetCustomAttribute(attributeHandle).Constructor;
+
+                        string attributeTypeName = null, attributeTypeNamespace = null;
+
+                        if (attributeCtor.Kind == HandleKind.MemberReference)
+                        {
+                            EntityHandle attributeMemberParent = mdReader.GetMemberReference((MemberReferenceHandle)attributeCtor).Parent;
+                            if (attributeMemberParent.Kind == HandleKind.TypeReference)
+                            {
+                                TypeReference attributeTypeRef = mdReader.GetTypeReference((TypeReferenceHandle)attributeMemberParent);
+                                attributeTypeName = mdReader.GetString(attributeTypeRef.Name);
+                                attributeTypeNamespace = mdReader.GetString(attributeTypeRef.Namespace);
+                            }
+                        }
+                        else if (attributeCtor.Kind == HandleKind.MethodDefinition)
+                        {
+                            var attributeTypeDefHandle = (TypeDefinitionHandle)mdReader.GetMethodDefinition((MethodDefinitionHandle)attributeCtor).GetDeclaringType();
+                            TypeDefinition attributeTypeDef = mdReader.GetTypeDefinition(attributeTypeDefHandle);
+                            attributeTypeName = mdReader.GetString(attributeTypeDef.Name);
+                            attributeTypeNamespace = mdReader.GetString(attributeTypeDef.Namespace);
+                        }
+
+                        if (attributeTypeName == "ReferenceAssemblyAttribute" && attributeTypeNamespace == "System.Runtime.CompilerServices")
```

Comment:
You should use mdReader.StringComparer.Equals instead of calling GetString and comparing strings. This will save on unnecessary string allocations.